### DISCUSSION TITLE
JIT Compiler seems to limit some regular TailwindCSS functionality.

### DISCUSSION THREAD

**Discussion by OD91** [tailwindlabs/tailwindcss#5196] [Help]
Hi, 

I run into some problems when using the jit mode in my tailwind.config.js file

For Example:

- If I want to give a div a border and switch between colors: This works perfectly fine without the JIT but doesn't work with the JIT - I always have to restart the dev server for seeing the changes...
- If I have a ternary which sets a font weight to medium if true: this doesn't work with JIT, the font weight doesn't change - even not if I restart the dev server...

Can anyone help?

**Comment by simonswiss** [comment]
Hey!

If you haven't yet, please refer to the [troubleshooting section](https://tailwindcss.com/docs/just-in-time-mode#troubleshooting) of the Tailwind CSS docs.

If you have to constantly restart your server, adding `TAILWIND_MODE=watch` at the start of your dev script might help.

When it comes to the second point, it sounds like you're trying to do some dynamic classes. Here's an example:

```js
font-${props.isBold ? 'bold' : 'normal'}
```

This will **not work** with Tailwind CSS (JIT or not), because the mechanism to purge/generate styles needs to find the _entire class string_ in one bit.

Here's the relevant documentation section for that:

https://tailwindcss.com/docs/optimizing-for-production#writing-purgeable-html

![image](https://user-images.githubusercontent.com/485747/129317395-2064f21d-feb5-4550-835d-5f6198ba1fc5.png)

Hope it helps!

**Reply by OD91** [reply]
Hi Simon, thanks for your quick reply! The `TAILWIND_MODE=watch` helps (imho this isn't as easily described in the docs as you just did - so maybe you can update this) , the other one unfortunately does not help...
